docs/pages/demo/datepicker/AdvancedKeyboard.example.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { Fragment, useState } from 'react';
+import EventNoteIcon from '@material-ui/icons/EventNote';
+import { DesktopDatePicker } from '@material-ui/pickers';
+
+function AdvancedKeyboardExample(props) {
+  const [selectedDate, handleDateChange] = useState(new Date());
+
+  return (
+    <Fragment>
+      <DesktopDatePicker
+        autoOk
+        variant="outlined"
+        label="Advanced keyboard"
+        placeholder="2018/01/01"
+        format={props.__willBeReplacedGetFormatString({
+          moment: 'YYYY/MM/DD',
+          dateFns: 'yyyy/MM/dd',
+        })}
+        mask="____/__/__"
+        keyboardIcon={<EventNoteIcon />}
+        value={selectedDate}
+        onChange={date => handleDateChange(date)}
```

Comment:
We cannot use event in `onChange` because this feature can be fired from
1) keyboard input (we have event)
2) clicking `ok` button (we have event, but it is `MouseEvent` so I think we already cannot pass 2 different event types)
3) pressing enter with open datepicker (again different event type)
4) selecting date in the last available view <-- **Here** we don't have an event 

Again passing event will make more confusion